WhoIsThat/WhoIsThat/ViewModels/HomeViewModel.cs

```diff
@@ -22,85 +24,186 @@ public class HomeViewModel : INotifyPropertyChanged
 
-            //Checking for storage permissions
             var storagePermission = await PermissionHandler.CheckForCameraPermission();
             if (!storagePermission)
                 await CrossPermissions.Current.RequestPermissionsAsync(Permission.Storage);
+            try
+            {
+                var takenPhoto = await TakingPhotoHandler.TakePhoto();
 
-            //Taking photo and storing it in MediaFile variable 'takenPhoto'
-            var takenPhoto = await TakingPhotoHandler.TakePhoto();
-
-            //Save taken photo to Azure cloud for recognition, later on it is deleted
-            await CloudStorageService.SaveBlockBlob(takenPhoto,"temp.jpg");
+                await CloudStorageService.SaveBlockBlob(takenPhoto, "temp.jpg");
+            }
             
-            //Binding taken image for display
-            DisplayStream = ImageSource.FromStream(() =>
+            catch (ManagerException photoNotTakenException)
             {
-                var stream = takenPhoto.GetStream();
-                takenPhoto.Dispose();
-                return stream;
-            });
-            
-            OnPropertyChanged("DisplayStream");
+                DisplayStatus = photoNotTakenException.ErrorCode;
+                OnPropertyChanged("DisplayStatus");
 
-            //Initiating recognition API
-            var restService = new RestService();
-            var recognizedName = await restService.Identify();
+                return;
+            }
 
-            //Checking whether person was identified and deciding on messages to display
-            if (IsIdentified(recognizedName))
+            try
             {
-                DisplayMessage = "It's a match!";
+                var recognitionMessage = await _restService.Identify();
+                var isTargetDead = await _restService.IsPreyHunted(User.Id, Convert.ToInt32(recognitionMessage));
+
+                if (isTargetDead)
+                {
+                    DisplayMessage = "It's a direct hit!";
+                    OnPropertyChanged("DisplayMessage");
+
+                    DisplayStatus = "Please wait...";
+                    OnPropertyChanged("DisplayStatus");
+
+                    var hitResult = await _restService.GetUserById(Convert.ToInt32(recognitionMessage));
+
+                    DisplayMessage = "Get to know each other!";
+                    OnPropertyChanged("DisplayMessage");
+
+                    DisplayStatus = hitResult.PersonFirstName;
+                    OnPropertyChanged("DisplayStatus");
+
+                    User = await _restService.UpdateUserScore(User.Id);
+
+                    Name = "Welcome, " + User.PersonFirstName + ". Score: " + User.Score.ToString();
+                    OnPropertyChanged("Name");
+                }
+            }
+
+            catch (ManagerException noFacesFoundException) when (noFacesFoundException.ErrorCode == Constants.NoFacesIdentifiedError)
```

Comment:
You could probably shorten it up to `ex` :D

Reply:
I tried to make it look serious, but yeah, I could definitely do that :D